Automattic/jetpack
# Publicize: cache connection testing results during HTTP request

#### Changes proposed in this Pull Request:
Connection testing may get requested multiple times during the same HTTP request.
This PR stores the result in the object property to avoid excessive requests.

#### Other information:


#### Jetpack product discussion
p1664903339698629-slack-CDLH4C1UZ

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Before testing the PR, please go through the instructions on `trunk` first to confirm the issue.

1. Install [this plugin](https://gist.github.com/sergeymitr/69dcd087837fdf69892c926189370028) to track XMLRPC requests. _It doesn't exclusively track connection testing requests, so the number of requests logged may be a bit bigger than connection tests._
2. Activate Publicize and connect it to at least one social account.
1. Create a few posts scheduled for the next month.
3. Open the file `tools/docker/wordpress/wp-content/debug-publicize.log`, confirm it's not empty.
4. Create another post and start scheduling, switch to the next month.
5. Open the `debug-publicize.log`:
6.1. On `trunk` there'll be one request logged for each already scheduled post in that month, per connection. If you have 5 posts scheduled, you’ll see 5 requests for each connection.
6.2. With the branch checked out, there'll only be one request per connection, no matter how many posts already scheduled.
